safenode/src/node/api.rs

```diff
@@ -129,20 +129,28 @@ impl Node {
             NetworkEvent::RequestReceived { req, channel } => {
                 self.handle_request(req, channel).await?
             }
-            NetworkEvent::PeerAdded => {
+            NetworkEvent::PeerAdded(peer) => {
                 self.events_channel.broadcast(NodeEvent::ConnectedToNetwork);
-                let target = {
-                    let mut rng = rand::thread_rng();
-                    XorName::random(&mut rng)
-                };
-
+                let key = NetworkKey::from_peer(peer);
                 let network = self.network.clone();
                 let _handle = spawn(async move {
-                    trace!("Getting closest peers for target {target:?}");
-                    let result = network.node_get_closest_peers(target).await;
-                    trace!("For target {target:?}, get closest peers {result:?}");
+                    trace!("Getting closest peers for target {key:?}...");
+                    let result = network.node_get_closest_peers(&key).await;
+                    trace!("Closest peers to {key:?} got: {result:?}.");
                 });
             }
+            NetworkEvent::PeersAdded(peers) => {
+                self.events_channel.broadcast(NodeEvent::ConnectedToNetwork);
+                for peer in peers {
```

Comment:
the purpose of this get_closest block is just to get the node try to discover other nodes.
So, no need to `get_closest for each peer` , just pick one (or a buset, say number_nodes/8 ) random peer to get_closest shall be enough.

Reply:
Will fix!

Reply:
But how many could actually be gotten in such a vec of peers? Is it many many?

Reply:
Fixed.